File: java/day-9/codes/hibernate-project/src/main/java/com/wipro/hibernate/entities/Product.java
package com.wipro.hibernate.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Entity
@Table(name = "PRODUCT_DTLS")
@Data
@RequiredArgsConstructor //@NoArgsConstructor
@AllArgsConstructor
public class Product {
	@Id
	//@GeneratedValue(strategy = GenerationType.AUTO)
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long productId;
	private String productName;
	private Float productPrice;
	
	public Product(String productName, Float productPrice) {
		super();
		this.productName = productName;
		this.productPrice = productPrice;
	}	
}

File: day-21/17.react-app/src/components/Product.js
import React from 'react'

/*
const Product = (props) => {
  console.log(props)  
  return (    
        <tr key={props.product.productId}>
            <td>{props.product.productId}</td>
            <td>{props.product.productName}</td>
            <td>{props.product.productPrice.toFixed(2)}</td>
        </tr>
  )
}
  */

const Product = ({product}) => {
  console.log(product)  
  return (    
        <tr key={product.productId}>
            <td>{product.productId}</td>
            <td>{product.productName}</td>
            <td>{product.productPrice.toFixed(2)}</td>
        </tr>
  )
}

export default Product
